includes/Feed/FeedConfigurationDetection.php

```diff
@@ -0,0 +1,170 @@
+		$graph_api           = $integration->get_graph_api();
+		$integration_feed_id = $integration->get_feed_id();
+		$catalog_id          = $integration->get_product_catalog_id();
+
+		$info = array();
+		$info['site-feed-id']  = $integration_feed_id;
+
+		// No catalog id. Most probably means that we don't have a valid connection.
+		if ( '' === $catalog_id ) {
+			throw new Error( 'No catalog ID' );
+		}
+
+		// Get all feeds configured for the catalog.
+		$feed_nodes = $this->get_feed_nodes_for_catalog( $catalog_id, $graph_api );
+
+		$info['feed-count'] = count( $feed_nodes );
+
+		// Check if the catalog has any feed configured.
+		if ( empty( $feed_nodes ) ) {
+			throw new Error( 'No feed nodes for catalog' );
+		}
+
+		// We will only track settings for one feed config (for now at least).
+		// So we need to determine which is the most relevant feed.
+		// If there is only one, we use that.
+		// If one has the same ID as $integration_feed_id, we use that.
+		// Otherwise we pick the one that was most recently updated.
+		$active_feed_metadata = null;
+		foreach ( $feed_nodes as $feed ) {
+			$metadata = $this->get_feed_metadata( $feed['id'], $graph_api );
+
+			if ( $feed['id'] === $integration_feed_id ) {
+				$active_feed_metadata = $metadata;
+				break;
+			}
+
+			if ( ! array_key_exists( 'latest_upload', $metadata ) || ! array_key_exists( 'start_time', $metadata['latest_upload'] ) ) {
+				continue;
+			}
+			$metadata['latest_upload_time'] = strtotime( $metadata['latest_upload']['start_time'] );
+			if ( ! $active_feed_metadata ||
+				( $metadata['latest_upload_time'] > $active_feed_metadata['latest_upload_time'] ) ) {
+				$active_feed_metadata = $metadata;
+			}
+		}
+
+		$active_feed['created-time']  = gmdate( 'Y-m-d H:i:s', strtotime( $active_feed_metadata['created_time'] ) );
+		$active_feed['product-count'] = $active_feed_metadata['product_count'];
+
+		// Upload schedule settings can be in two keys:
+		// `schedule` => full replace of catalog with items in feed (including delete).
+		// `update_schedule` => append any new or updated products to catalog.
+		// These may both be configured; we will track settings for each individually (i.e. both).
+		// https://developers.facebook.com/docs/marketing-api/reference/product-feed/
+		if ( array_key_exists( 'schedule', $active_feed_metadata ) ) {
+			$active_feed['schedule']['interval']       = $active_feed_metadata['schedule']['interval'];
+			$active_feed['schedule']['interval-count'] = $active_feed_metadata['schedule']['interval_count'];
+		}
+		if ( array_key_exists( 'update_schedule', $active_feed_metadata ) ) {
+			$active_feed['update-schedule']['interval']       = $active_feed_metadata['update_schedule']['interval'];
+			$active_feed['update-schedule']['interval-count'] = $active_feed_metadata['update_schedule']['interval_count'];
+		}
+
+		$info['active-feed'] = $active_feed;
+
+		$latest_upload      = $active_feed_metadata['latest_upload'];
```

Comment:
It looks like this may be problematic. I have created a new fresh feed config. It immediately started to fetch the data. I have killed the local server to simulate upload failure. I hoped for an empty 'latest_upload' array but still, I got some data:

![image](https://user-images.githubusercontent.com/17271089/120312411-f7242e00-c2d8-11eb-8eb4-5840f9a9fb0e.png)

Start and end times are the same indicating that nothing really has happened.

This means that even if this is broken on a merchant's site we will not catch that. This is true for this part and for the code at line 93 for detecting the lastest upload time.

Reply:
I have dig through the documentation and I have found nothing that would directly indicate that a feed upload is valid or not. One approach is to request uploads instead of 'latest_upload' and parse data in there.
Still, we are not able to get clear direct information but it looks like there is a solid way to get what we want.
We need to request additional fields:
`xxx/?fields=uploads{start_time,end_time,num_detected_items,num_deleted_items,num_invalid_items,num_persisted_items}`

When the feed upload fails the fields:
`num_detected_items,num_deleted_items,num_invalid_items,num_persisted_items` all equal to 0. When the feed upload is a success then some of the fields are not equal to 0:
![image](https://user-images.githubusercontent.com/17271089/120318008-93e9ca00-c2df-11eb-8fb9-3afb496787e2.png)

There is also an errors field but it would be very problematic to parse it for this information because it can have an for the whole feed or error field for each individual line item. It just has textual information without any sort of code.

The `latest_uploads` does not have these additional fields.  If we want to just check the `latest_upload` but with the additional data we can do that by limiting the `uploads` field:
`?fields=uploads.limit(1){start_time,end_time,num_detected_items,num_deleted_items,num_invalid_items,num_persisted_items,errors{id,suggested_rules}}`

We need to get that right since we expect that most of the sites do not have a usable feed.

Reply:
> This means that even if this is broken on a merchant's site we will not catch that. This is true for this part and for the code at line 93 for detecting the lastest upload time.

> We need to request additional fields: `fields=uploads{start_time,end_time,num_detected_items,num_deleted_items,num_invalid_items,num_persisted_items}

Yes - this is what I'm thinking too! I'm using this kind of approach to populate `latest-upload`. 

https://github.com/woocommerce/facebook-for-woocommerce/blob/15fd70979f82e2e66b91b655413efc406c6bebc0/includes/Feed/FeedConfigurationDetection.php#L121-L125

I'm thinking that we'll use a combination of the following fields to get an idea of whether feed is working or not. For example, for your test data (feed not working):

```
active-feed.product-count = 8 
active-feed.latest-upload.num-persisted-items = 0
active-feed.latest-upload.error-count = 1
```

If the feed is working correctly, `error-count` will be zero, and `num-persisted-items` will be non-zero; ideally, equal to `product-count`.

Note: The goal of this PR is to track enough data to help us release a new feed engine with confidence. In future we may need to build on this to detect and migrate existing config - but that's a different goal. Do you think the data collected here will give enough insight for our launch plan? If there's something important missing, let's add it.